onnx/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/api/inference/onnx/classification/ImageRecognitionModel.kt

```diff
@@ -44,3 +46,37 @@ public open class ImageRecognitionModel(
         (internalModel as OnnxInferenceModel).initializeWith(*executionProviders)
     }
 }
+
+/**
+ * Predicts object for the given [imageProxy].
+ * Internal preprocessing is updated to rotate image to match target orientation.
+ * After prediction, internal preprocessing is restored to the original state.
+ *
+ * @param [imageProxy] Input image.
+ *
+ * @return The label of the recognized object with the highest probability.
+ */
+public fun ImageRecognitionModel.predictObject(imageProxy: ImageProxy): String {
+    val currentRotation = targetRotation
+    targetRotation = imageProxy.imageInfo.rotationDegrees
+    return predictObject(imageProxy.toBitmap()).also { targetRotation = currentRotation }
+}
+
+/**
+ * Predicts [topK] objects for the given [imageProxy].
+ * Internal preprocessing is updated to rotate image to match target orientation.
+ * After prediction, internal preprocessing is restored to the original state.
+ *
+ * @param [imageProxy] Input image.
+ * @param [topK] Number of top ranked predictions to return
+ *
+ * @return The list of pairs <label, probability> sorted from the most probable to the lowest probable.
+ */
+public fun ImageRecognitionModel.predictTopKObjects(
+    imageProxy: ImageProxy,
+    topK: Int = 5
+): List<Pair<String, Float>> {
+    val currentRotation = targetRotation
```

Comment:
Seems like a common pattern. Do you think it would make sense to have a function like this
```kotlin
public fun <R> CameraXCompatibleModel.doWithRotation(rotation: Int, function: () -> R): R {
    val currentRotation = targetRotation
    targetRotation = rotation
    return function().apply { targetRotation = currentRotation }
}
```
?